hedera-node/src/main/java/com/hedera/services/store/models/TopicConversion.java

```diff
@@ -35,13 +35,29 @@ public static Topic fromMerkle(final MerkleTopic merkleTopic, final Id id) {
 		return modelTopic;
 	}
 
+	/**
+	 * <p>Maps the model topic's fields with the fields of the Merkle topic</p>
+	 * @param model
+	 * 		<p>The source model topic</p>
+	 * @param merkleTopic
+	<p>The targeted Merkle topic</p>
+	 */
+	public static void mapModelChangesToMerkle(final Topic model, final MerkleTopic merkleTopic) {
+		modelToMerkle(model, merkleTopic);
+	}
+
+	//TODO: Fix the failing test.
 	private static void merkleToModel(final MerkleTopic merkle, final Topic model) {
-		model.setAdminKey(merkle.getAdminKey());
-		model.setSubmitKey(merkle.getSubmitKey());
-		model.setMemo(merkle.getMemo());
-		if (merkle.getAutoRenewAccountId() != null) {
+		if (merkle.hasAdminKey()) {
+			model.setAdminKey(merkle.getAdminKey());
+		}
+		if (merkle.hasSubmitKey()) {
+			model.setSubmitKey(merkle.getSubmitKey());
+		}
+		if (merkle.hasAutoRenewAccountId()) {
```

Comment:
I think that we might have a problem with the `has` checks. If the merkle topic **does not have** admin key, the model will have `adminKey=null` value and once we get to the `persist` method, the `null` value will be persisted either way.

So regardless if we have a check for `hasAdminKey` or not, the model will have `null` admin key. We could remove the checks altogether and have the `null` values be set, propagated & persisted. 

I think that this is the better option, because **if** we have values that when they are not set are different than `null`, we will **not load them** and persist them as `null` afterwards.

Reply:
I tried the suggested approach 

![image](https://user-images.githubusercontent.com/77110630/135208462-f1ab9c21-5767-4b88-9b2e-16397d39c130.png)

and some tests are failing (clearingAdminKeyWhenAutoRenewAccountPresent() is failing due to the recent migration of the validation logic in the Topic.update() method. 

**With the suggested implementation:**

![image](https://user-images.githubusercontent.com/77110630/135208366-955eac55-bc80-4cea-83f0-62fc13b7e4e1.png)

**When reverting to the current implementation:**

![image](https://user-images.githubusercontent.com/77110630/135208701-6c05c8ae-2b57-47d3-ad0d-af38e39e753d.png)